src/com/amazon/corretto/crypto/provider/AesGcmSpi.java

```diff
@@ -860,8 +920,8 @@ protected synchronized int engineUpdate(ByteBuffer input, ByteBuffer output) thr
     }
 
     @Override
-    protected int engineDoFinal(ByteBuffer input, ByteBuffer output) throws ShortBufferException,
-        IllegalBlockSizeException, BadPaddingException
+    protected int engineDoFinal(final ByteBuffer input, final ByteBuffer output)
```

Comment:
sorry, missed this earlier. should this be `synchronized` as well?

Reply:
Doesn't actually matter because it calls into other methods which synchronize.

I think that I'm going to go through and neaten up the synchronization logic since we've evolved so much since it was first written.

Reply:
I've gone back to the *original* version of this file and found why the synchronization is there. It predates the existence of `NativeContext` (with built in locking). Since we lock at the context level now in all cases (and that's the only state we keep around) we can remove all other synchronization from this class.

Incorrect multi-threaded code is very slightly more likely to result in an exception now (rather than undefined behavior which isn't an exception) but that code was never safe or correct to begin with.

Reply:
so we don't need to worry about e.g. race conditions around `decryptInputBuf.write` in multi-threaded [calls to `engineUpdate`](https://github.com/corretto/amazon-corretto-crypto-provider/pull/204/files/adc1300861f02dec5011d5edbc45e34f3d271b0d..60fc618c8e96682329db9c94719137375f3b0883#diff-b98fb0cf29aa3c6687048b3876d672e7d6d3840dd3200b6a2bd775b6eb84785eR439)? the caller is responsible for that synchronization?